pkg/innerring/processors/netmap/process_peers.go

```diff
@@ -142,3 +145,70 @@ func (np *Processor) processUpdatePeer(ev netmapEvent.UpdatePeer) {
 		np.log.Error("can't invoke netmap.UpdatePeer", zap.Error(err))
 	}
 }
+
+func (np *Processor) processRemoveSubnetNode(ev subnetEvent.RemoveNode) {
+	if !np.alphabetState.IsAlphabet() {
+		np.log.Info("non alphabet mode, ignore remove node from subnet notification")
+		return
+	}
+
+	candidates, err := np.netmapClient.GetCandidates()
+	if err != nil {
+		np.log.Warn("could not get network map candidates",
+			zap.Error(err),
+		)
+		return
+	}
+
+	removeFromSubnetwork := ev.SubnetworkID()
+
+	for _, node := range candidates.Nodes {
+		if !bytes.Equal(node.NodeInfo.PublicKey(), ev.Node().Bytes()) {
+			continue
+		}
+
+		hasOtherSubnets := false
+
+		err = node.IterateSubnets(func(subNetID subnetid.ID) error {
+			if subNetID.Equals(removeFromSubnetwork) {
+				return netmap.ErrRemoveSubnet
+			}
+
+			hasOtherSubnets = true
```

Comment:
`node.IterateSubnets` returns error in this case. We can use `if-else` below with `err != nil` condition.

Reply:
changed